cfme/tests/control/test_compliance.py

```diff
@@ -77,75 +74,49 @@ def assign_policy_for_testing(policy_for_testing, host, policy_profile_name):
-            # If something was happening, restart and wait for the UI to reappear to prevent errors
-            appliance.ipapp.reboot()
-            pytest.skip(
-                "Error during appliance configuration. Skipping:\n{}: {}".format(
-                    type(e).__name__, str(e)))
-        vm = VM.factory(appl_name, provider)
-    except exceptions.VMNotFoundViaIP:
-        logger.info("Provisioning a new appliance on provider %s.", provider.key)
-        appliance = provision_appliance(
-            vm_name_prefix=PREFIX + "host_",
-            version=str(version.current_version()),
-            provider_name=provider.key)
-        request.addfinalizer(lambda: diaper(appliance.destroy))
-        try:
-            appliance.configure(setup_fleece=True)
-        except (EOFError, ApplianceException) as e:   # Add known exceptions as needed.
-            pytest.skip(
-                "Error during appliance configuration. Skipping:\n{}: {}".format(
-                    type(e).__name__, str(e)))
-        vm = VM.factory(appliance.vm_name, provider)
-    if provider.type in {"rhevm"}:
-        request.addfinalizer(appliance.remove_rhev_direct_lun_disk)
-    # Do the final touches
-    with appliance.ipapp(browser_steal=True) as appl:
-        appl.set_session_timeout(86400)
-        provider.refresh_provider_relationships()
-        vm.wait_to_appear()
-        vm.load_details()
-        wait_for_ssa_enabled()
-        yield vm
+def configure_fleecing(appliance, has_no_providers_modscope, provider, setup_provider_modscope):
+    setup_providers_hosts_credentials(provider.key)
+    appliance.install_vddk(reboot=False, vddk_url=vddk_url_map[str(provider.version)])
+    appliance.reboot(quit_browser=True)
+    yield
+    appliance.uninstall_vddk()
+
+
+@pytest.yield_fixture(scope="module")
+def compliance_vm(configure_fleecing, provider, full_template_modscope):
+    name = "{}-{}".format("test-compliance", fauxfactory.gen_alpha(4))
+    vm = VM.factory(name, provider, template_name=full_template_modscope["name"])
+    vm.create_on_provider(allow_skip="default")
+    provider.mgmt.start_vm(vm.name)
+    provider.mgmt.wait_vm_running(vm.name)
+    if not vm.exists:
+        vm.wait_to_appear(timeout=900)
+    yield vm
+    if provider.mgmt.does_vm_exist(vm.name):
+        provider.mgmt.delete_vm(vm.name)
+    provider.refresh_provider_relationships()
 
 
 @pytest.yield_fixture(scope="module")
-def analysis_profile(compliance_vm):
-    ap = AnalysisProfile(name="default", description="ap-desc", profile_type='VM', files=[],
-                         categories=["check_software"])
+def analysis_profile():
+    ap = AnalysisProfile(
+        name="default",
+        description="ap-desc",
+        profile_type="VM",
+        categories=[
+            "check_services",
+            "check_accounts",
+            "check_software",
+            "check_vmconfig",
+            "check_system"
+        ]
+    )
     if ap.exists:
```

Comment:
**REQUIRED**: We discussed that this is a slightly too magical and should be using just the standard create method in the fixture